doc/source/cluster/key-concepts.rst

```diff
@@ -60,19 +60,31 @@ sit idle, the autoscaler will remove worker nodes from the cluster.
 It is important to understand that the autoscaler only reacts to task and actor resource requests, and not application metrics or physical resource utilization.
 To learn more about autoscaling, refer to the user guides for Ray clusters on :ref:`VMs <cloud-vm-index>` and :ref:`Kubernetes <kuberay-index>`.
 
+
+.. _ray_driver:
+
+Ray Driver
+----------
+
+A Ray driver is the process that calls the Ray tasks and actors in the user application.
+A driver is started by running `ray.init()`. The script that calls `ray.init()` may also be referred to as the "driver script".
+
+
 .. _cluster-clients-and-jobs:
 
-Ray Jobs
---------
+Ray Job
```

Comment:
What are the names of the two types of jobs. One is `Ray Job Submission` and what's the other one?

Reply:
Yea, then do we need a noun called `Ray Job Submission` if it's just a way to run a job.

Or should it be `Submission Job` instead of `Job Submission` to make it clear that it's still a job.

Reply:
Ray job submission -> the submission of the job. It depicts this behavior instead of the job itself.
To sum up,
- There is only 1 job concept: Ray job.
- There are 3 ways (run driver on any node, ray client, submit jobs via job api) to run a Ray job
- There is one api called Job API for submitting jobs to remote clusters, listing jobs, retrieving logs, etc.

Is my understanding aligned with yours? @rkooo567 @architkulkarni @stephanie-wang @jjyao

Reply:
yeah this sounds good.

Reply:
This sounds good to me.  I'll update this PR accordingly

Reply:
Yea, that aligns with my understanding as well.